pcp/PCPMachine.c

```diff
@@ -50,40 +50,97 @@ static void PCPMachine_updateCPUcount(PCPMachine* this) {
    this->values = xCalloc(cpus, sizeof(pmAtomValue));
 }
 
-static void PCPMachine_updateMemoryInfo(Machine* host) {
+static void PCPMachine_updateSystemName(PCPMachine* this) {
+   pmAtomValue sysname;
+   if (!Metric_values(PCP_UNAME_SYSNAME, &sysname, 1, PM_TYPE_STRING))
+      sysname.cp = NULL;
+   else if (String_eq(sysname.cp, "Linux"))
+      this->sys = SYSTEM_NAME_LINUX;
+   else if (String_eq(sysname.cp, "Darwin"))
+      this->sys = SYSTEM_NAME_DARWIN;
+   free(sysname.cp);
+}
+
+static void PCPMachine_updateLinuxMemoryInfo(PCPMachine* this) {
```

Comment:
Ditto. (Consider reusing the `Platform_setMemoryValues` code in `linux/Platform.c` and don't write everything twice (WET).)